### PR TITLE
Imp wave2 861 imp fgs compare programs

### PR DESCRIPTION
**Pull request by mhughes2012**
WIP for review by @dpfitzsi

### CODE REVIEW

**Comment by dpfitzsi** [comment]
Hi Mike, can you take another look at the latest code on this branch? I think all of the filtering is now fixed but you can verify against the AC.

File: sites/all/modules/ucalgary/ucgy_fgs/includes/rs_filters.inc [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+              <button type="button" class="btn btn-default course-based" value="course-based">Course based</button>
+              <button type="button" class="btn btn-default thesis-based" value="thesis-based">Thesis based</button>
+            </p>
+          </div>
+          <div class="col-md-5 vspacer-xs vspacer-sm faculty">
+            <p><?php print t('Filter by faculty'); ?>:</p>
+            <select id="faculty-filter" autocomplete="off" class="selectpicker" title="View all faculties" data-width="auto">
+              <option value="all" selected><?php print t('View all faculties'); ?></option>
+              <option data-divider="true"></option>
+              <?php foreach($faculties as $faculty): ?>
```
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_fgs/includes/rs_filters.inc]: Where is $faculties being set?

Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/includes/rs_filters.inc]: Thanks @robbdavis. Dan and I discussed this, but it hadn't been done until now. I've added the reference to it $faculties = entity_load_multiple_by_name('faculty');
and activated the filter in the markup.

**Comment by robbdavis** [comment]
Thanks Mike.

The filtering and view displaying is working. If that is all this story is focusing on then I am approving.

The javascript, however, is messed up. The graduate and undergraduate compare javascripts are conflicting with each other. And the behavior to remove a degree is not working either. 

As long as those things will be addressed in another story then this is approved.

File: sites/all/modules/ucalgary/ucgy_fgs/includes/rs_filters.inc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            --><span class="label label-default thesis-based">Thesis based<span class="glyphicon glyphicon-remove" aria-hidden="true"></span></span><!--
+            Faculty filter appears here --></div>
+          <div id="view-count"><!-- Program count appears here --></div>
+        </div>
+
+      </div>
+      <div class="row">
+        <div class="col-sm-12 compare">
+          <p>
+            <?php if (empty($fgs_degrees)): ?>
```
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_fgs/includes/rs_filters.inc]: I don't think $fgs_degrees is set anywhere?

File: sites/all/modules/ucalgary/ucgy_fgs/includes/rs_filters.inc [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+              <?php print t('We are experiencing technical issues. Please try again in a few moments. Thank you for your continued patience.'); ?>
```
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_fgs/includes/rs_filters.inc]: This is getting printed out even if there are results.

File: sites/all/modules/ucalgary/ucgy_fgs/templates/page--future-students--graduate--program-compare.tpl.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
+<?php if ($logged_in && $is_admin) { ?><!-- Include file: <?php echo str_replace('-', '=', __FILE__); ?> --><?php } ?>
+<?php
+$module_http = '/' . drupal_get_path('module','ucgy_fgs');
```
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/page--future-students--graduate--program-compare.tpl.php]: Why is this here? Does this variable get called somewhere?

File: sites/all/modules/ucalgary/ucgy_fgs/templates/page--future-students--undergraduate--program-compare.tpl.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
+<?php if ($logged_in && $is_admin) { ?><!-- Include file: <?php echo str_replace('-', '=', __FILE__); ?> --><?php } ?>
```
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/page--future-students--undergraduate--program-compare.tpl.php]: Why is this template for undergraduate--program-comper.tpl.php in the ucgy_fgs feature?

File: sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.info [hunk 6]
```diff
@@ -27,6 +27,8 @@ dependencies[] = view_unpublished
 stylesheets[all][] = css/ucgy_fgs_graduate_degree.css
 dependencies[] = views
 stylesheets[all][] = css/ucgy_fgs_graduate_degree.css
+scripts[] = js/comparison.js
```
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.info]: Just a note that this will load the javascript on every page. Including the undergraduate pages.
Also, we are already loading this file from ucgy_fs.info (or at least a js with the same name). We should not be loading it in both places.

File: sites/all/modules/ucalgary/ucgy_fgs/includes/rs_filters.inc (hunk 2 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/includes/rs_filters.inc]: I change this.

File: sites/all/modules/ucalgary/ucgy_fgs/includes/rs_filters.inc (hunk 3 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/includes/rs_filters.inc]: I've changed this check to check against the view results count

File: sites/all/modules/ucalgary/ucgy_fgs/templates/page--future-students--graduate--program-compare.tpl.php (hunk 4 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/page--future-students--graduate--program-compare.tpl.php]: Removed.

File: sites/all/modules/ucalgary/ucgy_fgs/templates/page--future-students--undergraduate--program-compare.tpl.php (hunk 5 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/page--future-students--undergraduate--program-compare.tpl.php]: Removed.

File: sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.info (hunk 6 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.info]: Removed. file removed as well since it already exists in fs

File: sites/all/modules/ucalgary/ucgy_fgs/templates/views-view--fgs-program-comparison.tpl.php [hunk 7]
```diff
@@ -0,0 +1,99 @@
+<?php
+//kpr('views-view--program-requirements--block.tpl');
```
Comment (dpfitzsi) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/views-view--fgs-program-comparison.tpl.php]: Can you remove this debug code?

File: sites/all/modules/ucalgary/ucgy_fgs/templates/views-view--fgs-program-comparison.tpl.php [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+    <?php if ($feed_icon): ?>
+        <div class="feed-icon">
+            <?php print $feed_icon; ?>
+        </div>
+    <?php endif; ?>
+
+</div><?php /* class view */ ?>
+
+<?php
```
Comment (dpfitzsi) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/views-view--fgs-program-comparison.tpl.php]: More debug code that isn't running. This can be removed.

File: sites/all/modules/ucalgary/ucgy_fgs/templates/views-view--fgs-program-comparison.tpl.php (hunk 7 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/views-view--fgs-program-comparison.tpl.php]: removed debug code.

File: sites/all/modules/ucalgary/ucgy_fgs/templates/views-view--fgs-program-comparison.tpl.php (hunk 8 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/views-view--fgs-program-comparison.tpl.php]: removed debug code here too.

**Comment by balrajb** [comment]
question: in probo https://eb4a6540-b4be-4b88-b5a5-8c74f9195fc4--pr-468.probo.build/future-students/undergraduate/program-compare?compare=1867+1940+150+184 is 404 , it should work like http://develop-ucalgary.pantheonsite.io//future-students/undergraduate/program-compare?compare=1867+1940+150+184
